tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+                                  "\n/google-cloud-support change-priority [case number] [priority, e.g. P1] -- changes the priority of the case"
+                                  #"\n/google-cloud-support escalate [case number] [reason,           REASON_UNSPECIFIED|RESOLUTION_TIME|TECHNICAL_EXPERTISE|BUSINESS_IMPACT] [justification] -- escalates the case, contact your TAM if you encounter issues when trying to escalate the case"
+                                  "\n/google-cloud-support stop-tracking [case number] -- case updates will no longer be posted to this channel"
+                                  #"\n/google-cloud-support close-case [case number] -- closes the case and stops tracking it for all channels"
+                                  "\n/google-cloud-support list-tracked-cases -- lists all cases being tracked in this channel"
+                                  "\n/google-cloud-support list-tracked-cases-all -- lists all cases being tracked in the workspace"
+                                  "\n/google-cloud-support case-details [case_number] -- pull all of the case deta as json"
+                                  "\n/google-cloud-support sitrep -- report of all active cases in the org")
+
+    return Response(), 200
+
+
+
+def track_case(channel_id, channel_name, case, user_id):
+    """
+    Add a Google Cloud support case to a list of tracked cases. If the case can't be
+    found in the list of active support cases, notify the user.
+
+    Parameters
+    ----------
+    channel_id : str
+        unique string used to idenify a Slack channel. Used to send messages to the channel
+    channel_name : str
+        designated channel name of the channel. For users to understand where their
+        cases are being tracked in Slack
+    case : str
+        unique id of the case
+    user_id : str
+        the Slack user_id of the user who submitted the request. Used to send ephemeral
+        messages to the user
+    """
+    
+    parent = get_parent(case)
+    if parent == 'Case not found':
+        client.chat_postEphemeral(channel=channel_id, user=user_id, text=f"Case {case} could not be found in your org. If this case was recently created, please give the system 60 seconds to fetch it. Otherwise, double check your case number or confirm the org being tracked with your Slack admin.")
+    else:
+        tracker = {
+            "channel_id": channel_id, 
+            "case": case,
+            "channel_name": channel_name
+        }
+
+        exists = False
+    
+        for tc in tracked_cases:
+            if tc['channel_id'] == channel_id and tc['case'] == case:
+                exists = True
+                break
+
+        if exists == False:
+            tracked_cases.append(tracker)
+            
+            try:
+                file_overwrite(tracked_cases_file, tracked_cases)
+            except Exception as e:
```

Comment:
What specific exception are you capturing here? Please narrow it down to the specific situations that you are trying to handle, this is too broad.